modules/explorer/info.go

```diff
@@ -38,6 +38,17 @@ func (e *Explorer) BlockFacts(height types.BlockHeight) (modules.BlockFacts, boo
 	return bf.BlockFacts, true
 }
 
+// LatestBlockFacts returns a set of statistics about the blockchain as they appeared
+// at the latest block height in the explorer's consensus set.
+func (e *Explorer) LatestBlockFacts() (modules.BlockFacts, bool) {
+	var height types.BlockHeight
+	err := e.db.View(dbGetInternal(internalBlockHeight, &height))
+	if err != nil {
+		return modules.BlockFacts{}, false
+	}
+	return e.BlockFacts(height)
```

Comment:
don't call an exported function from within an exported function, the concurrency conventions we use can result in deadlocks by doing that.

Within an exported function, you can safely assume full control of a mutex, assume that nobody else holding one has called you, etc. `defer Unlock` is a common pattern, and a person deciding to call that shouldn't be frustrated by having to deal with another exported call from inside the function, if at all possible.